src/omegga/plugin.ts

```diff
@@ -345,9 +351,112 @@ export class PluginLoader {
 
+    let loadOrder: Plugin[] = [];
+
+    if (unsortedPlugins.length) {
+      // create a DAG of plugins based on dependencies, loadBefore, and loadAfter
+      const depGraph = unsortedPlugins.reduce(
+        (graph, plugin) => {
+          const name = plugin.getName();
+          const {
+            dependencies = {},
+            loadAfter = [],
+            loadBefore = [],
+          } = plugin.pluginConfig ?? {};
+
+          graph[name] ??= { } as any;
+          graph[name].plugin ??= plugin;
+          graph[name].behind ??= [];
+          graph[name].inDegree ??= 0;
+
+          // add edges for everything ahead of name
+          for (const depName of [...loadAfter, ...Object.keys(dependencies)]) {
+            graph[name].behind.push(depName);
+
+            graph[depName] ??= { } as any;
+            graph[depName].inDegree ??= 0;
+            graph[depName].inDegree++;
+          }
+
+          // add edges for everything behind name
+          for (const depName of [...loadBefore]) {
+            graph[depName] ??= { } as any;
+            graph[depName].behind ??= [];
+            graph[depName].behind.push(name);
+
+            graph[name].inDegree++;
+          }
+
+          return graph;
+        },
+        {} as {
+          [pluginName: string]: {
+            plugin: Plugin;
+            inDegree: number;
+            behind: string[];
+          };
+        },
+      );
+
+      // find all nodes with no incoming edges to start processing
+      let zeros: string[] = Object.entries(depGraph).reduce(
+        (arr, [name, { inDegree }]) => {
+          if (inDegree === 0) {
+            arr.push(name);
+          }
+          return arr;
+        },
+        [],
+      );
+
+      // execute a topological sort, grouping nodes where the order doesn't matter, in the order that *does* matter
+      const result: string[][] = [];
+      while (zeros.length) {
+        // these nodes can be in any order
+        result.unshift([...zeros]);
+
+        const nextZero = [];
+
+        for (const node of zeros) {
+          for (const neighbor of depGraph[node].behind) {
+            depGraph[neighbor].inDegree--;
+            if (depGraph[neighbor].inDegree === 0) {
+              nextZero.push(neighbor);
+            }
+          }
+        }
+
+        zeros = nextZero;
+      }
+
+      if (Object.values(depGraph).some(({ inDegree }) => inDegree > 0)) {
+        Logger.verbose(
+          'Cyclic or missing dependencies detected in plugins. Could not resolve load order.',
+        );
+        return false;
+      }
+
+      // sort each group where the order doesn't matter by loadPriority and then alphabetically
+      // then flatten the result so now we have a well defined load order
+      loadOrder = result
+        .map(interchangeablePlugins =>
+          interchangeablePlugins
+            // map it back to a plugin
+            .map(name => depGraph[name].plugin)
+            .sort((a, b) => {
```

Comment:
shouldn't this load order happen at the beginning? It looks like the load order built by the graph is wiped by this .sort